src/g3log/loglevels.hpp

```diff
@@ -22,6 +22,15 @@
 #endif
 #endif
 
+#if (defined _MSC_VER || defined __GCC__)
+  #ifdef g3log_EXPORTS
+    #define g3log_API __declspec(dllexport)
```

Comment:
I thought there was a CMake way of exporting symbols without "polluting" the source code with Windows specific defines in front of the API, have you checked into that?